9548 saml soap backchannel logout

lagess opened the pull request:
Closes #9548

pedroigor committed:
Allow updating authz settings via default client registration provider

Closes #9008

pedroigor committed:
Allow managing the required settigs for the email attribute

Closes #15026

martin-kanis committed:
Replace old HotRod index annotation with new one

Commit:
SOAP backchannel logout for SAML protocol (#9548)

mposolda committed:
Make Keycloak FIPS working with OpenJDK 17 on FIPS enabled RHEL
Closes #15721

Commit:
return SOAP Fault if issue in SOAP binding

manusa committed:
ci: bump actions-setup-minikube to v2.7.2

The current version might have trouble with recent
versions of Minikube. Especially since the latest
release of crictl (this morning).

Relates to:
- https://github.com/kubernetes-sigs/cri-tools/releases/tag/v1.26.0
- https://github.com/manusa/actions-setup-minikube/releases/tag/v2.7.2

pedroigor committed:
Make it possible to set a custom user and cache providers when using legacy store

Closes #15008

ahus1 committed:
Normalize JVM heap usage in tests and handle OOM situations

Closes #16089

ahus1 committed:
Ensure lock table has its primary key created, and re-enable the DBLockTest

Closes #15487

stianst committed:
Upload Surefire reports to Foresight (#16013)

ahus1 committed:
Removing testsuite/performance from main Keycloak repository (#15950)

There's a separate repository keycloak-benchmark for this.

Closes #14192

stianst committed:
Fix paths to surefire-reports for Foresight (#16019)

pedroigor committed:
Use git checkout to rever changes

Closes #15644

klausbetz committed:
improve german email translation (#15949)

Closes #15948

stianst committed:
Flaky test reporter (#16034)

* Flaky test reporter

* Update .github/actions/upload-flaky-tests/action.yml

stianst committed:
Use test instead of install to run tests, and optimize unit tests job (#16006)

stianst committed:
Add language injection for larger bash scripts in composite actions (#16040)

mhajas committed:
Move HotRod profile to the same pom as other map profiles and introduce map-storage-chm profile

Closes #16046

ahus1 committed:
Adding CRDB into GHA for the new store (#16021)

The CockroachDB database is slower than PostgreSQL, therefore it will only run branches and nightly builds.

Closes #16020

dependabot[bot] committed:
Bump actions/setup-java from 2 to 3

Bumps [actions/setup-java](https://github.com/actions/setup-java) from 2 to 3.
- [Release notes](https://github.com/actions/setup-java/releases)
- [Commits](https://github.com/actions/setup-java/compare/v2...v3)

---
updated-dependencies:
- dependency-name: actions/setup-java
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

martin-kanis committed:
Upgrade to Infinispan 14.0.4.Final

basbeu commented on services/src/main/java/org/keycloak/protocol/saml/profile/util/Soap.java:
```diff
@@ -110,6 +111,19 @@ public static Document extractSoapMessage(SOAPMessage soapMessage) {
+     * Get the first direct child that is an XML element
+     * (avoid returning text in case the first child is preceded by some new line)
+     */
+    public static Node getFirstChild(Node parent) {
```
I did not get the purpose of this method at the first read. Maybe the comment should more clear
And it seems to be not tested by a unit test

lagess replied:
comment improved

basbeu commented on services/src/main/java/org/keycloak/protocol/saml/SamlProtocol.java:
```diff
@@ -748,6 +759,38 @@ protected Response buildLogoutResponse(UserSessionModel userSession, String logo
+                    throw new RuntimeException("Logout response received from client is not an StatusResponseType");
+                }
+                StatusResponseType logoutStatus = (StatusResponseType)samlDoc.getSamlObject();
+                if (!logoutStatus.getStatus().getStatusCode().getValue().equals(JBossSAMLURIConstants.STATUS_SUCCESS.getUri())) {
```
Is NPE impossible here ?

lagess replied:
Could be indeed, but if there is an NPE, then the client has provided a response that is not valid according to the SAML protocol. I've slightly adapted the equality check to be a bit safer, and to do the same as an existing piece of code (https://github.com/keycloak/keycloak/blob/168734b817da2138548bacba973204e80f566b2c/services/src/main/java/org/keycloak/broker/saml/SAMLEndpoint.java#L616)

vramik committed:
Generate getMapKeyClass and getMapValueClass methods for map fields

Closes #16053

stianst committed:
Upload flaky test reports to artifacts (#16076)

dependabot[bot] committed:
Bump github/codeql-action from 2.1.36 to 2.1.37 (#16050)

Bumps [github/codeql-action](https://github.com/github/codeql-action) from 2.1.36 to 2.1.37.
- [Release notes](https://github.com/github/codeql-action/releases)
- [Changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md)
- [Commits](https://github.com/github/codeql-action/compare/v2.1.36...v2.1.37)

---
updated-dependencies:
- dependency-name: github/codeql-action
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

abstractj committed:
Update to Quarkus 2.13.6.Final

Resolves #16084

ahus1 committed:
Enable retryable transactions for CockroachDB model tests

Also made the test setup of the realm more idempotent to clean up a realm left-over from a previous test.

Closes #16078

hmlnarik committed:
Don't print warning when heapdump not found

Fixes: #16125

miquelsi committed:
Added Oracle to Legacy Store IT. (#16097)

stianst committed:
Suspend scheduled tasks if time offset is set to a large value (#16144)

dependabot[bot] committed:
Bump jquery

Bumps [jquery](https://github.com/jquery/jquery) from 3.6.1 to 3.6.3.
- [Release notes](https://github.com/jquery/jquery/releases)
- [Commits](https://github.com/jquery/jquery/compare/3.6.1...3.6.3)

---
updated-dependencies:
- dependency-name: jquery
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

stianst committed:
Run Keycloak CI every 3 hours during Christmas holidays (#16149)
